### ISSUE TITLE
Allow configuring client API key using external file

### ISSUE THREAD

**Issue by zaytsev** [sigoden/aichat#802]
**Is your feature request related to a problem? Please describe.**
I'm using [sops-nix](https://github.com/Mic92/sops-nix) to manage all sensitive info like credentials, keys, etc. By design sops-nix stores secrets in individual files, which is fine for most cases. But for aichat i had to create a workaround

**Describe the solution you'd like**

It would be nice to have a config property to set client API key via external file, e.g. `api_key_file`

**Describe alternatives you've considered**

I ended up with a shell wrapper
```
export CLIENT1_API_KEY = `cat {client1_api_key_file}`
export CLIENT2_API_KEY = `cat {client2_api_key_file}`
export CLIENT3_API_KEY = `cat {client3_api_key_file}`

/path/to/aichat
```

**Additional context**

**Comment by sigoden** [comment]
We will not support this feature, please use env file instead, see #685

**Comment by sigoden** [comment]
Sorry, No.

I'm not familiar with `sops-nix`, but if it can use a file to store a single `api_key` value, why can't it use a file to store the contents of a `.env` file?

**Comment by zaytsev** [comment]
Right, technically i can keep all my API keys as key-value pairs in a single sops entry. But then it makes declarative config kind of obscure and harder to manage, so instead of explicit

```nix
{config, ...}: {
  programs.aichat = {
    enable = true;
    config = {
      compress_threshold = 6000;
      clients = {
        openai = {
          enable = true;
          api_key_file = config.sops.secrets.openai_api_key.path;
        };
        claude = {
          enable = true;
          api_key_file = config.sops.secrets.anthropic_api_key.path;
        };
      };
    };
  };
}
```
I will have to remember to update yet another config file.

Anyway, thanks for creating and improving aichat!

**Comment by 5HT2** [comment]
> @sigoden We will not support this feature, please use env file instead, see #685

Hi, would you please reconsider the ability to provide an API key provider or a shell command to provide one?

All of my secrets / keys are stored inside of a local keychain, and require a hardware form of 2fa to access them, I (additionally) have my API tokens setup to rotate or require a re-authorization depending on the API tokens, I would very much like to be able to manage my credentials externally without just providing a (commonly-targeted) `.env` file exposed on my filesystem.

Alternatively, would you be willing to consider a pull request to discuss the implementation of this / the semantics of how this would look on the users end and the code changes required? It should not be too big of a change.

**Comment by sigoden** [comment]
Sorry, we will not support reading the API key from an external program.
